UI/src/view/AssetFormFactory.js

```diff
@@ -952,29 +894,121 @@
       return sideCode ? _assetTypeConfiguration.className + '-' + sideCode : _assetTypeConfiguration.className;
     }
 
-    function validateAdministrativeClass(selectedLinearAsset, editConstrains){
-      editConstrains = editConstrains || function() { return false; };
+    function checkEditConstrains(selectedAsset){
+      var editConstrains = _assetTypeConfiguration.editConstrains || function() { return false; };
 
-      var selectedAssets = _.filter(selectedLinearAsset.get(), function (selected) {
-        return editConstrains(selected);
+      var selectedAssets = _.filter(selectedAsset.get(), function (asset) {
+        return editConstrains(asset);
       });
       return !_.isEmpty(selectedAssets);
     }
 
-    var renderLinktoWorkList = function renderLinktoWorkList(layerName) {
-      var textName;
-      switch(layerName) {
-        case "maintenanceRoad":
-          textName = "Tarkistamattomien huoltoteiden lista";
-          break;
-        default:
-          textName = "Vanhentuneiden kohteiden lista";
+    me.isSplitOrSeparatedAllowed = function(){
+      var numberFields = function(sideCode) {
+        return _.flatten(_.map(forms.getFields(sideCode) , function(property) {
+          return property.getPropertyValue().value;
+        })).length;
+      };
+
+      if(_.isEmpty(forms.getAllFields()))
+        return false;
+
+      if (numberFields('a') !== numberFields('b'))
+        return true;
+
+      return _.some(forms.getFields('a'), function(fieldA){
+        var propertyValueA = fieldA.getPropertyValue();
+
+        var propertyValueB = _.head(_.map(_.filter(forms.getFields('b'), function (fieldB) {
```

Comment:
First you should add a new function on the dynamicField called getPublicId() then what you want here is
```
var fieldB = _.find(forms.getFields('b'), function(field) { return field.getPublicId() == propertyValueA.publicId; } );
if(!fieldB)
  return true;
return !fieldA.compare(propertyValueA, fieldB.getPropertyValues()));
```